Clang tidy: Replace typdef with using decleration

DavidFair opened the pull request:
**Description of work.**
Replaces typedefs with using. The latter is much more readable (being
closer to an assignment), and having a single style makes it more the
codebase more maintainable

**To test:**
- Check tests pass

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

DavidFair committed:
Clang tidy: Replace typdef with using decleration

Replaces typedefs with using. The latter is much more readable (being
closer to an assignment), and having a single style makes it more the
codebase more maintainable

cailafinn reviewed (changes requested):
Looks good, and much nicer to be switching to `using`. I noticed that there still seem to be a couple of `typedef`s in `MockScriptRepository.h` and `BaseCustomInstrumentPresenter.h`. Do they also need to be switched out?

DavidFair committed:
Manually replace remaining instances

Clang-tidy missed some instances, so manually replace these using a
regex instead.

cailafinn reviewed (approved):
Looks good to me now :)
